### ISSUE TITLE
Support user auth with GetFederationToken

### ISSUE THREAD

**Issue by cyounkins** [Miserlou/Zappa#1218]
## Context

Some corporate environments use AWS Security Token Service to deal out temporary credentials to developers. Comparison of STS methods: http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_request.html#stsapi_comparison

Note that if you give out temporary credentials using GetFederationToken, the assumed role cannot call any IAM methods. IAM:GetRole is required by Zappa even with "manage_roles": false, and this causes it to fail.

## Expected Behavior

I expect `update` to work in my federated environment.

## Actual Behavior

Zappa tries to call GetRole which fails.

```
$ zappa update production
Calling update for stage production..
Downloading and installing dependencies..
Packaging project as zip.
Uploading image-viewer-production-1509656439.zip (12.9MiB)..
...
Updating Lambda function code..
Updating Lambda function configuration..
Oh no! An error occurred! :(

==============

Traceback (most recent call last):
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/zappa/cli.py", line 2525, in handle
    sys.exit(cli.handle())
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/zappa/cli.py", line 484, in handle
    self.dispatch_command(self.command, stage)
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/zappa/cli.py", line 528, in dispatch_command
    self.update()
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/zappa/cli.py", line 887, in update
    aws_kms_key_arn=self.aws_kms_key_arn
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/zappa/core.py", line 1019, in update_lambda_configuration
    self.get_credentials_arn()
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/zappa/core.py", line 2009, in get_credentials_arn
    self.credentials_arn = role.arn
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/boto3/resources/factory.py", line 339, in property_loader
    self.load()
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/boto3/resources/factory.py", line 505, in do_action
    response = action(self, *args, **kwargs)
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/cyounkins/code/av/source/offboard/labeler/tools/elastic_labeling/image_viewer/env/lib/python2.7/site-packages/botocore/client.py", line 601, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (InvalidClientTokenId) when calling the GetRole operation: The security token included in the request is invalid
```

## Possible Fix

The traceback indicates that Zappa is trying the resolve a human-readable role name (optionally provided as role_name in settings) to an ARN. I suggest the settings file be able to accept the ARN directly.

## Steps to Reproduce

1.
2.
3.

## Your Environment

* Zappa version used:
* Operating System and Python version:
* The output of `pip freeze`:
* Link to your project (optional):
* Your `zappa_settings.py`: 

Zappa 0.44.3
Settings has "manage_roles": false

**Comment by Miserlou** [comment]
Are you supplying a `aws_kms_key_arn`?

**Comment by cyounkins** [comment]
No, not supplying `aws_kms_key_arn` and I'm not sure why I would. I don't have control over the distribution of temporary credentials. I can however use special admin to configure the IAM role as needed.

**Comment by Miserlou** [comment]
Gotcha. Can you generate an aws credential profile from the info you have?
Obviously I don't have very much context for your environment and
application here, so it's possible this is a feature that you'll need to
implement yourself. Should be an easy one. PRs welcome!

**Comment by cyounkins** [comment]
Fixed by #1219
